Source/WebCore/rendering/updating/RenderTreeUpdaterGeneratedContent.cpp

```diff
@@ -263,4 +263,50 @@ void RenderTreeUpdater::GeneratedContent::removeAfterPseudoElement(Element& elem
     element.clearAfterPseudoElement();
 }
 
+void RenderTreeUpdater::GeneratedContent::updateWritingSuggestionsRenderer(RenderElement& renderer)
+{
+    auto destroyWritingSuggestionsIfNeeded = [&renderer, this]() {
+        if (WeakPtr writingSuggestionsRenderer = renderer.writingSuggestionsRenderer())
+            m_updater.m_builder.destroy(*writingSuggestionsRenderer);
+    };
+    if (!renderer.element()) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (renderer.element()->writingSuggestion().isEmpty()) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (!renderer.canHaveChildren())
+        return;
+
+    auto style = renderer.getCachedPseudoStyle({ PseudoId::InternalWritingSuggestions }, &renderer.style());
+    if (!style || style->display() == DisplayType::None) {
+        destroyWritingSuggestionsIfNeeded();
+        return;
+    }
+
+    if (renderer.writingSuggestionsRenderer() && renderer.writingSuggestionsRenderer()->nextSibling())
+        destroyWritingSuggestionsIfNeeded();
+
+    auto newStyle = RenderStyle::clone(*style);
+    if (auto writingSuggestionsRenderer = renderer.writingSuggestionsRenderer()) {
+        writingSuggestionsRenderer->setStyle(WTFMove(newStyle));
+
+        if (auto child = writingSuggestionsRenderer->firstChild())
+            uncheckedDowncast<RenderText>(child)->setText(renderer.element()->writingSuggestion());
+    } else {
+        auto newWritingSuggestionsRendererParent = WebCore::createRenderer<RenderInline>(RenderObject::Type::Inline, renderer.document(), WTFMove(newStyle));
+        newWritingSuggestionsRendererParent->initializeStyle();
+
+        auto newWritingSuggestionsRenderer = WebCore::createRenderer<RenderText>(RenderObject::Type::Text, renderer.document(), renderer.element()->writingSuggestion());
+        m_updater.m_builder.attach(*newWritingSuggestionsRendererParent, WTFMove(newWritingSuggestionsRenderer));
+
+        renderer.setWritingSuggestionsRenderer(*newWritingSuggestionsRendererParent.get());
```

Comment:
There is a confusion here between what is a "writingSuggestionsRenderer" and what is a "writingSuggestionsRendererParent". You should make this consistent, this does not need the extra confusion.

Reply:
renamed the variables to be more clear